libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -58,6 +58,9 @@ void PostTypeChecker::endVisit(ContractDefinition const&)
 	for (auto declaration: m_constVariables)
 		if (auto identifier = findCycle(declaration))
 			typeError(declaration->location(), "The value of the constant " + declaration->name() + " has a cyclic dependency via " + identifier->name() + ".");
+
+	m_constVariables.clear();
```

Comment:
I wonder whether it might be better to remove the check in line 51. The reason is that I think it is safer to rather keep the dependencies in. Furthermore, we already plan to allow things like
```js
library X { uint constant x = 0x123; }
contract C {
uint y = X.x;
}
```
and thus we might have cross-contract dependencies soon.